tests/integration/test_paths.py

```diff
@@ -191,6 +191,13 @@ def test_replace_file(ws, make_random, cls):
         tmp_dir.rmdir(recursive=True)
 
 
+@pytest.mark.parametrize("cls", DATABRICKS_PATHLIKE)
+def test_resolve_is_consistent(ws, cls):
+    path = cls(ws, "/a/b/c") / Path("../../d")
+    resolved = path.resolve()
+    assert resolved == cls(ws, "/a/d")
```

Comment:
From a semantics perspective, the following also holds and should be tested:

 * `/a/b/c/../../d` → `/a/d`
 * `/../b` → `/b`

A quirk of Posix is that the root directory has a `..` entry that points to itself. This behaviour has been taken over into normalisation for URLs, as described [here](https://datatracker.ietf.org/doc/html/rfc3986#section-5.2.4). We should do the same.

This example from that reference (similar to mine above) would also be an excellent example to test:

 * `/a/b/c/./../../g` → `/a/g`

Reply:
done